cmd/compute-domain-controller/daemonset.go

```diff
@@ -228,6 +229,20 @@ func (m *DaemonSetManager) Create(ctx context.Context, namespace string, cd *nva
 	return d, nil
 }
 
+func (m *DaemonSetManager) Get(ctx context.Context, cdUID string) (*appsv1.DaemonSet, error) {
+	ds, err := getByComputeDomainUID[*appsv1.DaemonSet](ctx, m.mutationCache, cdUID)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DaemonSet: %w", err)
+	}
+	if len(ds) > 1 {
+		return nil, fmt.Errorf("more than one DaemonSet found with same ComputeDomain UID")
+	}
+	if len(ds) == 0 {
+		return nil, nil
+	}
+	return ds[0], nil
+}
+
 func (m *DaemonSetManager) Delete(ctx context.Context, cdUID string) error {
 	ds, err := getByComputeDomainUID[*appsv1.DaemonSet](ctx, m.mutationCache, cdUID)
```

Comment:
Question -- could we not replace this code block with a call to `m.Get(ctx, string(cd.UID))`?